cli,kv,server: consolidate `StatusClient` RPC client creation

cthumuluru-crdb opened the pull request:
This commit consolidates `StatusClient` RPC client creation logic in cli, kv, and server packages. It is a continuation of the work done in #147606.

Epic: CRDB-48923
Fixes: https://github.com/cockroachdb/cockroach/issues/148099
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148011)
<!-- Reviewable:end -->

cthumuluru-crdb committed:
cli,kv,server: consolidate `StatusClient` RPC client creation

This commit consolidates `StatusClient` RPC client creation logic in
cli, kv, and server packages. It is a continuation of the work done in
[#147606].

Epic: CRDB-48923
Fixes: #148099
Release note: none

tbg reviewed (approved):
Looks good. If there are parts that merit a close review, please give me a heads up - I mostly scrolled through and checked that the changes all fit in with what I was expecting.

arjunmahishi reviewed (approved):
The tsdump and zip changes look good to me!

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148011#-:-OSSRVlIAa9C4jqtMihy:b-2zizei)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, @dhartunian, and @tbg)



<!-- Sent from Reviewable.io -->

aa-joshi reviewed (approved):
zip and tsdump changes  <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 23 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148011#-:-OSSS-8171WZIvUEuQ2D:b-9mtakx)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @Abhinav1299, @dhartunian, and @tbg)



<!-- Sent from Reviewable.io -->

cthumuluru-crdb commented on pkg/cli/tsdump.go:
```diff
@@ -138,13 +139,15 @@ will then convert it to the --format requested in the current invocation.
 			defer finish()
 
-			names, err := serverpb.GetInternalTimeseriesNamesFromServer(ctx, conn)
+			target, _ := addr.AddrWithDefaultLocalhost(serverCfg.AdvertiseAddr)
```
`target, _ := addr.AddrWithDefaultLocalhost(serverCfg.AdvertiseAddr)` - target was taken from the connection before. I extracted these bits out to build the target. @aa-joshi, @arjunmahishi, I'm assuming it is fine. Right?

cthumuluru-crdb reviewed (commented):
@tbg - mostly the changes in `pkg/kv/kvserver/loqrecovery/server.go` and `pkg/server/fanout\_clients.go`. I replaced the direct dial with a dialer. I ensure the dialer is created only once per visitor.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148011#-:-OSSVH1hA45nN1cmlnny:bei8vgz)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, @arjunmahishi, @dhartunian, and @tbg)



<!-- Sent from Reviewable.io -->

cthumuluru-crdb commented:
bors r+
